### PR TITLE
[HOPSWORKS-2024] fix consul hostname resolution on ubuntu containers

### PR DESCRIPTION
**Pull request by SirOibaf**

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -17,13 +17,8 @@
                     kubernetes_domain_name = node['kube-hops']['cluster_domain']
                 end
             end
-
-        end
-        if node['install']['localhost'].casecmp?("true")
-            my_ip = "127.0.0.1"
```
Comment (kouzant) [inline recipes/default.rb:23]: This variable is used in another places in the recipe

File: recipes/default.rb [hunk 2]
```diff
@@ -48,15 +43,15 @@
...
                     ip6tables-save | sudo tee /etc/iptables/rules.v6
-                    sed -i "s/#DNS=/DNS=127.0.0.2/g" /etc/systemd/resolved.conf
+                    sed -i "s/#DNS=/DNS=#{my_private_ip()}/g" /etc/systemd/resolved.conf
                     sed -i "s/#Domains=/Domains=~#{node['consul']['domain']}/g" /etc/systemd/resolved.conf
                 EOH
             end
 
             if node['install']['localhost'].casecmp?("true")
-                dnsmasq_ip = "127.0.0.2"
+                dnsmasq_ips = "127.0.0.2,127.0.0.1"
```
Comment (kouzant) [inline recipes/default.rb:52]: Why do we need both 127.0.0.2 and 127.0.0.1 ? I guess we can squash them. I think `my_private_ip()` returns 127.0.0.1 when localhost==true

**Review by kouzant — CHANGES_REQUESTED** [review]

File: recipes/default.rb (hunk 1 above)
Comment (SirOibaf) [reply recipes/default.rb:23]: My bad

File: recipes/default.rb (hunk 2 above)
Comment (SirOibaf) [reply recipes/default.rb]: Squashed.

**Comment by tkakantousis** [comment]
Tests passed, LGTM

**Review by kouzant — APPROVED** [review]

**Pull request merged**